core/tools/implementations/runTerminalCommand.ts

```diff
@@ -1,34 +1,128 @@
 import childProcess from "node:child_process";
 import util from "node:util";
 
-import { ToolImpl } from ".";
 import { fileURLToPath } from "node:url";
+import { ToolImpl } from ".";
 
 const asyncExec = util.promisify(childProcess.exec);
 
 export const runTerminalCommandImpl: ToolImpl = async (args, extras) => {
   const ideInfo = await extras.ide.getIdeInfo();
+  const toolCallId = extras.toolCallId || "";
 
   if (ideInfo.remoteName === "local" || ideInfo.remoteName === "") {
-    try {
-      const output = await asyncExec(args.command, {
-        cwd: fileURLToPath((await extras.ide.getWorkspaceDirs())[0]),
+    // For streaming output
+    if (extras.onPartialOutput) {
+      return new Promise((resolve, reject) => {
+        try {
+          const getWorkspaceDirsPromise = extras.ide.getWorkspaceDirs();
+          getWorkspaceDirsPromise
+            .then((workspaceDirs) => {
+              const cwd = fileURLToPath(workspaceDirs[0]);
+              let stdout = "";
+              let stderr = "";
```

Comment:
```suggestion
              let terminalOutput = "";
```

We can use a single variable to aggreate results for both stdout, and stderr to handle commands that return both like cucumber.

You might want to also consider adding in a function to strip off ANSI terminal formatting to clean up what shows up in the partial, but retain the output into the actual tools output response. This seemed to work nicely for me.

```
// Function to strip ANSI escape sequences from terminal output
function stripAnsiCodes(str: string): string {
  // This regex matches ANSI escape codes used for terminal formatting
  return str.replace(/\x1B[[(?);]{0,2}(;?\d)*[@-~]/g, '');
}
```